tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/RabbitMQIntegration.cs

```diff
@@ -68,6 +68,52 @@ internal static Scope CreateScope(Tracer tracer, out RabbitMQTags tags, string c
             return scope;
         }
 
+        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IDictionary<string, object> headers)
+        {
+            var dataStreamsManager = tracer.TracerManager.DataStreamsManager;
+            if (dataStreamsManager == null || !dataStreamsManager.IsEnabled)
```

Comment:
Good point. Checkpoints without the context propagation are not really useful. I'll switch `RabbitMQHeadersCollectionAdapter ` to non-nullable headers and disable checkpointing if there no headers (Idk when this may happen, just followed the logic for context propagation for Spans). We can look into improving this in the future.